src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0034Tests.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Xunit;
+using Verifier = Azure.ClientSdk.Analyzers.Tests.AzureAnalyzerVerifier<Azure.ClientSdk.Analyzers.DuplicateTypeNameAnalyzer>;
+
+namespace Azure.ClientSdk.Analyzers.Tests
+{
+    public class AZC0034Tests
+    {
+        [Theory]
+        [InlineData("Azure.Data", "String", true)]
+        [InlineData("Azure.MyService", "BlobClient", true)]
+        [InlineData("Azure.MyService", "CosmosClient", true)]
+        [InlineData("MyCompany.Data", "String", false)]
+        public async Task AZC0034ProducedForReservedTypeNames(string namespaceName, string typeName, bool shouldReport)
+        {
+            var code = shouldReport 
+                ? $@"
+namespace {namespaceName}
+{{
+    public class {{|AZC0034:{typeName}|}} {{ }}
+}}"
+                : $@"
+namespace {namespaceName}
+{{
+    public class {typeName} {{ }}
```

Comment:
why do we need to test both just the typename and typename with the analyzer id?